### DISCUSSION TITLE
Windows Poetry Installation ERROR

### DISCUSSION THREAD

**Discussion by pravincoder** [python-poetry/poetry#9587] [Q&A]
### Description

Ran the Installation code of Poetry in powershell , Below is the result 

![Capture](https://github.com/user-attachments/assets/0a214e09-189b-4fe4-a596-b3049f88ec4a)

Now i believe that it partical install or there is some other issue because now my Device says poetry is UnRecognised , but when I Tried ReInstalling it is says it is already install 

Device Spec:- Windows 10, 64 bit 

pls , help me out guys , let me if there is a work around or alternative approach to install poetry

### Workarounds

I could'nt find any Work arounds

### Poetry Installation Method

install.python-poetry.org

### Operating System

Window 10

### Poetry Version

1.8.3

### Poetry Configuration

```bash session
none
```

### Poetry Runtime Logs

```bash session
PS C:\Users\WIN10> poetry --version
poetry : The term 'poetry' is not recognized as the name of a cmdlet, function, script file, or operable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:1
+ poetry --version
+ ~~~~~~
    + CategoryInfo          : ObjectNotFound: (poetry:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException
```

**Comment by finswimmer** [comment]
Hey @pravincoder,

as the installer told you, you have to add Poetry's bin directory once to `Path` environment variable. (Also have a look into the docs at step 3: https://python-poetry.org/docs/#installing-with-the-official-installer)

Here's an how-to if you don't know already how to set environment variables on windows: https://www.tenforums.com/tutorials/121855-edit-user-system-environment-variables-windows.html

**Comment by pravincoder** [comment]
> as the installer told you, you have to add Poetry's bin directory once to `Path` environment variable. (Also have a look into the docs at step 3: https://python-poetry.org/docs/#installing-with-the-official-installer)

@finswimmer 
I did tried that out ! , but there is no improvement  
so i found a few workarounds, the first one worked for me , also i'll  try to find out the reason for the error 

 WorkAround :- 
1. Used Normal Pip Approach to install and it worked for me (Not Recommended) 
2. Using pipx follow the  [installation doc](https://pipx.pypa.io/stable/installation/)

**Comment by Secrus** [comment]
@pravincoder after adding to the PATH, you most likely have to also restart the terminal so that the new PATH is picked up.

**Comment by pravincoder** [comment]
> @pravincoder after adding to the PATH, you most likely have to also restart the terminal so that the new PATH is picked up.

@Secrus  Did'nt work! 😅

I believe the path generated by the installation is incorrect those are path1 :- 'C:\Users\WIN10\AppData\Roaming\Python\Scripts'
path2 :- 'C:\Users\WIN10\AppData\Roaming\pypoetry\venv\Scripts' I added both as Env Variable and ran powershell command for poetry but error persist

**Comment by Secrus** [comment]
@pravincoder can you manually verify that the file `C:\Users\WIN10\AppData\Roaming\Python\Scripts\poetry` exists?

**Comment by pravincoder** [comment]
@Secrus it does here is an ss ,

![wr](https://github.com/user-attachments/assets/9906d458-623c-4f7b-97fd-b62928e628d1)

Also , environment variable  ss, 

![ws](https://github.com/user-attachments/assets/0e5bd941-1262-4db5-8b8b-5391535bd54e)
tried both paths as mentioned in the installation doc.

**Comment by dimbleby** [comment]
as the installer told you at the very start of this thread, you need poetry's bin directory in your `PATH` environment variable

what you have instead done is added environment variables called `poetry` and `poetry2`.

please google "windows how to add to path" or similar, and close this out

**Reply by pravincoder** [reply]
I tried every way to find the bin path , but was'nt any location for a bin file , As a result i tried with the poetry file location
